Sample-01/src/app/components/nav-bar/nav-bar.component.html

```diff
@@ -24,26 +24,33 @@
           <li class="nav-item">
             <a routerLink="/" class="nav-link">Home</a>
           </li>
-          <li class="nav-item" *ngIf="auth.loggedIn">
+          <li class="nav-item" *ngIf="auth.isAuthenticated$ | async">
             <a routerLink="external-api" class="nav-link">External API</a>
           </li>
         </ul>
 
-          <ul class="navbar-nav d-none d-md-block">
-            <!-- Login button: show if NOT authenticated -->
-            <li class="nav-item" *ngIf="auth.loggedIn === false">
-              <button
-                id="qsLoginBtn"
-                class="btn btn-primary btn-margin"
-                (click)="auth.login()"
-              >
-                Log in
-              </button>
-            </li>
-            <!-- / Login button -->
+        <ul class="navbar-nav d-none d-md-block">
+          <!-- Login button: show if NOT authenticated -->
+          <li
+            class="nav-item"
+            *ngIf="(auth.isAuthenticated$ | async) === false"
+          >
+            <button
+              id="qsLoginBtn"
+              class="btn btn-primary btn-margin"
+              (click)="loginWithRedirect()"
```

Comment:
Some could find confusing that for some things we use a direct call on the service `auth.isAuthenticated$` and for others we define our own function `loginWithRedirect()` (that internally calls the service).

I think the more we show the same we're doing on the quickstart article, the better. The suggestion here is to use component templates and always reference the service directly.

Reply:
The difference is calling methods vs using observables.

Also, I think separating the quickstart and the sample more is a good thing. The quickstart can show a more condensed example of the component but for the sample, I think a more real-world representation is better.

> always reference the service directly

I don't always find that the best way, especially when your method calls have string parameters; sometimes the syntax just doesn't work.